src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -833,6 +833,38 @@ GrpAb: Z^3
     return domain(map_from_torusinvariant_cartier_divisor_group_to_torusinvariant_weil_divisor_group(v))
 end
 
+@doc raw"""
+    map_from_torusinvariant_cartier_divisor_group_to_class_group(v::AbstractNormalToricVariety)
+
+Return the map from the Cartier divisors to the class group
+of an abstract normal toric variety `v`.
+
+# Examples
+```jldoctest
+julia> p2 = projective_space(NormalToricVariety, 2)
+Normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> map_from_torusinvariant_cartier_divisor_group_to_class_group(p2)
+Map with following data
+Domain:
+=======
+Abelian group with structure: Z^3
+Codomain:
+=========
+Abelian group with structure: Z
+```
+"""
+@attr GrpAbFinGenMap function map_from_torusinvariant_cartier_divisor_group_to_class_group(v::AbstractNormalToricVariety)
+    # check input
+    if has_torusfactor(v)
+        throw(ArgumentError("Group of the torus-invariant Cartier divisors can only be computed if the variety has no torus factor"))
+    end
```

Comment:
Can this also be achieved with the following?
```suggestion
    @req has_torusfactor(v) "Group of the torus-invariant Cartier divisors can only be computed if the variety has no torus factor"
```